cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -159,7 +164,7 @@ private Future<CruiseControlResponse> getCruiseControlState(String host, int por
         });
     }
 
-    private void internalRebalance(String host, int port, String path, String userTaskId,
+    private void internalRebalance(Reconciliation reconciliation, String host, int port, String path, String userTaskId,
```

Comment:
We are not logging the response in the last "else" for the unexpected status code. Note that in case of auth errors you actually get HTML back, not JSON. I would simply log the response status code and body as string one time before checking the outcome. Same for getUserTaskStatus. Wdyt?

Reply:
I haven't logged there because we are failing the future and the exception will be clear in the stacktrace so I don't think it's useful adding more.

> Also, shouldn't be trace level?

If you enable TRACE on the operator you will get a huge amount of information, more than what you really want and mixed through them the CC API REST requests/responses so it would be hard to find. I saw that we are using DEBUG in the KafkaConnect API client and agreed it was the right log level imho.

Reply:
These responses are big and may fill up the available log space pretty quickly in case of high reconciliation frequency. You can always enable trace on CruiseControlApiImpl class only.

Reply:
If you log at the start of the response handling, is one log line vs multiple log lines as you have right now.

Reply:
> You can always enable trace on CruiseControlApiImpl class only.

That's true you can tweak the `050-ConfigMap-strimzi-cluster-operator.yaml` if are installing using the YAMLs but what if you use operator hub, I am not sure it's that simple.

> If you log at the start of the response handling, it is one log line vs multiple log lines as you have right now.

The logs needs the user task ID and the body and we are not handling all the HTTP status code or retrieving this info for all of them. I could have moved retrieving user task ID at the top but not the body because it's part of the body handler which is different for each status code, this is why I am duplicating the lines.

Reply:
Ok, still not sure about the logging level, but it's not a big deal.

Reply:
FYI, I just waited too few but it's reloaded automatically if you change the ConfigMap.

Reply:
> @scholzj you already approved the PR, wdyt about this change? or you think we should stick with DEBUG without any additional stuff around ConfigMap and loggers?

I do not have a strong opinion about DEBUG versus TRACE because I never saw the actual messages. I guess maybe you should share some samples. 

But `These responses are big` sounds like it should not be logged at all? Wouldn't the responses from CC include like thousands of partitions etc.? I think at DEBUG level it would be a good idea to log what is being done and things such as return codes. But dumping full requests or responses for Cruise Control into log sounds like it would just screw up all other logs and not really help with Cruise Control debugging either. But if, then it should be probably at TRACE level?

I would not add anything special for this to the ConfigMap. We do not have it for other pieces as far as I remember.

Reply:
If you are looking for some kind of precedent, we never log the full Kubernetes resources in the operator classes when managing them in the Kubernetes cluster. We log only what we do and the outcome of the operation at DEBUG level. But not the resource itself.

Reply:
Looks good. I actually found useful to log the full response at trace level when working on RF change, but maybe it's not necessary here.

Reply:
Not sure about the RF change, but the full response could be really big here even because it's related to the movement of all partitions of all topics across all brokers (or some of them), which should not be the case for RF change.

Reply:
Yeah, it's usually shorter with RF.